### PR TITLE
feat: improvements in siwx flow

### PR DESCRIPTION
**Pull request by zoruka**
# Description

- Add SIWX handler for account switch
- Clean up SIWX initialization
- Fixes for multichain switches

## Type of change

- [ ] Chore (non-breaking change that addresses non-functional tasks, maintenance, or code quality improvements)
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Associated Issues

For Linear issues: Closes APKT-1320
For GH issues: closes #...

# Showcase (Optional)

If there is a UI change include the screenshots with before and after state.
If new feature is being introduced, include the link to demo recording.

# Checklist

- [ ] Code in this PR is covered by automated tests (Unit tests, E2E tests)
- [ ] My changes generate no new warnings
- [ ] I have reviewed my own code
- [ ] I have filled out all required sections
- [ ] I have tested my changes on the preview link
- [ ] Approver of this PR confirms that the changes are tested on the preview link

### CODE REVIEW

File: packages/adapters/solana/src/client.ts [hunk 1]
```diff
@@ -477,14 +477,14 @@ export class SolanaAdapter implements ChainAdapter {
         provider.chains.find(chain => chain.caipNetworkId === caipNetworkId) || provider.chains[0]
 
       if (connectionChain) {
+        ProviderUtil.setProvider(this.chainNamespace, provider)
+        this.provider = provider
+
```
Comment (tomiir) [inline packages/adapters/solana/src/client.ts:482]: why was this needed?

File: packages/appkit/src/universal-adapter/client.ts [hunk 2]
```diff
@@ -246,12 +247,35 @@ export class UniversalAdapterClient {
...
+          signature = (response as { signature: string }).signature
+        } else {
+          signature = await provider.request(
+            {
+              method: 'personal_sign',
+              params: [message, address]
+            },
+            ChainController.state.activeCaipNetwork?.caipNetworkId
+          )
+        }
```
Comment (tomiir) [inline packages/appkit/src/universal-adapter/client.ts:276]: hmmmm won't we need an if for every chain here? 
should we have a unified `signMessage` method that abstract this somehow ?

File: packages/adapters/solana/src/client.ts (hunk 1 above)
Comment (zoruka) [reply packages/adapters/solana/src/client.ts]: It was a leftover attempt to keep states synced as the provider was being get as undefined at some point due to the async task on network switch. I'm going to revert it.

File: packages/appkit/src/universal-adapter/client.ts (hunk 2 above)
Comment (zoruka) [reply packages/appkit/src/universal-adapter/client.ts:276]: Theoretically this should have been handled by the Solana Adapter, but when Solana adapter is not added as an adapter option it will use UPA and this case was not being handled correctly, that's why I had to add this conditional. If there is different implementation for each network then it UPA will need to handle all of them. I think the way to do it would have these methods implemented somewhere else and then call these methods from this function, what do you think?

**Review by tomiir — APPROVED** [review]
